delete.go

```
@@ -27,63 +27,74 @@ func killContainer(container *libcontainer.Container) error {
+Where "<container-id>" is the name(s) of the container(s) to delete.
 
 EXAMPLE:
-For example, if the container id is "ubuntu01" and runc list currently shows the
-status of "ubuntu01" as "stopped" the following will delete resources held for
-"ubuntu01" removing "ubuntu01" from the runc list of containers:
+For example, if the container ids are "ubuntu01" and "nginx01", and both are stopped, the following will delete resources for both containers:
 
-       # runc delete ubuntu01`,
+       # runc delete ubuntu01 nginx01`,
 	Flags: []cli.Flag{
 		cli.BoolFlag{
 			Name:  "force, f",
-			Usage: "Forcibly deletes the container if it is still running (uses SIGKILL)",
+			Usage: "Forcibly deletes the container(s) if still running (uses SIGKILL)",
 		},
 	},
 	Action: func(context *cli.Context) error {
-		if err := checkArgs(context, 1, exactArgs); err != nil {
+		if err := checkArgs(context, 1, minArgs); err != nil {
 			return err
 		}
 
-		id := context.Args().First()
 		force := context.Bool("force")
-		container, err := getContainer(context)
-		if err != nil {
-			if errors.Is(err, libcontainer.ErrNotExist) {
-				// if there was an aborted start or something of the sort then the container's directory could exist but
-				// libcontainer does not see it because the state.json file inside that directory was never created.
-				path := filepath.Join(context.GlobalString("root"), id)
-				if e := os.RemoveAll(path); e != nil {
-					fmt.Fprintf(os.Stderr, "remove %s: %v\n", path, e)
+		var errs []error
+
+		for _, id := range context.Args() {
+			container, err := getContainerByID(context, id)
+			if err != nil {
+				if errors.Is(err, libcontainer.ErrNotExist) {
+					// if container directory exists but no state.json, remove manually
+					path := filepath.Join(context.GlobalString("root"), id)
+					if e := os.RemoveAll(path); e != nil {
+						fmt.Fprintf(os.Stderr, "remove %s: %v\n", path, e)
+						errs = append(errs, e)
```

This removes the "`--force` ignores errors" case.

---

Yes, I used AI to help me get this done. I still made sure to review and test everything myself.